shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -50,82 +48,89 @@ bool IsAsciiPrintableKey(char ch) {
 
 TextInputChannel::TextInputChannel(
     BinaryMessenger* messenger,
-    std::unique_ptr<TizenInputMethodContext> input_method_context)
+    TizenInputMethodContext* input_method_context)
     : channel_(std::make_unique<MethodChannel<rapidjson::Document>>(
           messenger,
           kChannelName,
           &JsonMethodCodec::GetInstance())),
-      input_method_context_(std::move(input_method_context)) {
+      input_method_context_(input_method_context) {
   channel_->SetMethodCallHandler(
       [this](const MethodCall<rapidjson::Document>& call,
              std::unique_ptr<MethodResult<rapidjson::Document>> result) {
         HandleMethodCall(call, std::move(result));
       });
+}
 
-  // Set input method callbacks.
-  input_method_context_->SetOnPreeditStart([this]() {
-    FT_LOG(Debug) << "onPreeditStart";
-    active_model_->BeginComposing();
-  });
-
-  input_method_context_->SetOnPreeditChanged(
-      [this](std::string str, int cursor_pos) -> void {
-        FT_LOG(Debug) << "onPreedit: str[" << str << "] cursor_pos["
-                      << cursor_pos << "]";
-        if (str == "") {
-          // Enter pre-edit end stage.
-          return;
-        }
-        active_model_->UpdateComposingText(str);
-        SendStateUpdate(*active_model_);
-      });
+TextInputChannel::~TextInputChannel() {}
 
-  input_method_context_->SetOnPreeditEnd([this]() {
-    FT_LOG(Debug) << "onPreeditEnd";
+void TextInputChannel::OnComposeBegin() {
+  if (active_model_ == nullptr) {
+    return;
+  }
+  active_model_->BeginComposing();
+}
 
-    // Delete preedit-string, it will be committed.
-    int count = active_model_->composing_range().extent() -
-                active_model_->composing_range().base();
+void TextInputChannel::OnComposeChanged(const std::string& str,
+                                        int cursor_pos) {
+  if (active_model_ == nullptr) {
```

Comment:
Prefer if (value) over if (value != nullptr). [link](https://github.com/flutter-tizen/plugins/issues/354)
So. how about `if (!active_model)`?

Reply:
I always prefer `if (value)` than `if (value != nullptr)`.
but in this case, I prefer `if (value == nullptr)` than `if (!value)`, because it is used consistently, at least in this file, and the meaning is conveyed more clearly in situations where error handling is required.